contracts/src/v0.8/automation/dev/v2_3/AutomationRegistry2_3.sol

```diff
@@ -315,6 +315,7 @@ contract AutomationRegistry2_3 is AutomationRegistryBase2_3, OCR2Abstract, Chain
       delete s_signers[signerAddress];
       // Do not delete the whole transmitter struct as it has balance information stored
       s_transmitters[transmitterAddress].active = false;
+      s_deactivatedTransmittersList.push(transmitterAddress);
```

Comment:
a couple things here:
* This is only necessary if the payment mode is offchain, right? We can skip this step if payment mode is on-chain :)
* the way setConfig works if by deleting all previous transmitters, and then overwriting with the new ones. But many of the old transmitters will probably be a part of the new list. Ex say we do this...

    ```
    oldTransmitters = [A, B, C] 
    new transmitters = [A, B, D]

    what we want after setConfig():
    transmitters = [A, B, D]
    deactivated = [C]

    What we're going to have instead:
    transmitters = [A, B, D]
    deactivated = [A, B, C] // not what we want!
    ```
* I think if we change `s_deactivatedTransmittersList` to an `EnumerableSet`, we can more easily accomplish this. WDYT?